### PR TITLE
Implement readBytesAsStream() and writeBytesFromStream()  in tcp Client

### PR DESCRIPTION
**Pull request by MohamedSabthar**

### CODE REVIEW

File: tcp-ballerina/byte_stream.bal [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (BuddhiWathsala) [inline tcp-ballerina/byte_stream.bal:1]:
```suggestion
// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: tcp-ballerina/client.bal [hunk 2]
```diff
@@ -44,20 +44,58 @@ public client class Client {
...
+        Error? writeError = ();
+        while(true) {
+            record {|byte[] value;|}? bytes = iterator.next();
+            if (bytes is record {|byte[] value;|}) {
+                check self->writeBytes(bytes.value);
+                if (writeError is Error) {
+                    break;
+                }
+            } else {
+                _ = dataStream.close();
```
Comment (BuddhiWathsala) [inline tcp-ballerina/client.bal:65]: Shouldn't we need to return this error?

File: tcp-ballerina/tests/client_tests.bal [hunk 3]
```diff
@@ -75,6 +75,43 @@ function testServerAlreadyClosed() returns  @tainted error? {
...
+@test:Config {
+    dependsOn: [testServerAlreadyClosed]
+}
+function testReadBytesAsStream() returns  @tainted error? {
+    Client socketClient = check new ("localhost", PORT4);
+
+    Error|(stream<readonly & byte[]>) res = socketClient->readBytesAsStream();
+    if (res is stream<readonly & byte[]>) {
+        error? e = res.forEach(function (readonly & byte[] bytes) {
+                io:println(getString(bytes));
```
Comment (BuddhiWathsala) [inline tcp-ballerina/tests/client_tests.bal:87]: Shouldn’t we need to assert this?

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: tcp-ballerina/byte_stream.bal (hunk 1 above)
Comment (MohamedSabthar) [reply tcp-ballerina/byte_stream.bal:1]: updated

File: tcp-ballerina/client.bal (hunk 2 above)
Comment (MohamedSabthar) [reply tcp-ballerina/client.bal:65]: yes, updated with return

File: tcp-ballerina/tests/client_tests.bal (hunk 3 above)
Comment (MohamedSabthar) [reply tcp-ballerina/tests/client_tests.bal:87]: added `test:assertFail(msg = e.message());` in if

**Review by Maninda — APPROVED** [review]

**Comment by BuddhiWathsala** [comment]
@MohamedSabthar it seems, my previous suggestions are not addressed properly? Please recheck.

**Comment by MohamedSabthar** [comment]
> @MohamedSabthar it seems, my previous suggestions are not addressed properly? Please recheck.

Updated now.

**Comment by shafreenAnfar** [comment]
Closing this PR as we decided not to include this for swan lake release.

**Pull request closed without merging**